vision_agent_tools/models/florence2_sam2.py

```diff
@@ -4,16 +4,34 @@
 import torch
 import numpy as np
 from PIL import Image
-from pydantic import validate_call
-
-from vision_agent_tools.shared_types import BaseMLModel, VideoNumpy, SegmentationBitMask
+from pydantic import BaseModel, validate_call, Field
+from typing import Optional
+
+from vision_agent_tools.shared_types import (
+    BaseMLModel,
+    VideoNumpy,
+    SegmentationBitMask,
+    Device,
+)
 from vision_agent_tools.models.florencev2 import Florencev2, PromptTask
 
 from sam2.sam2_video_predictor import SAM2VideoPredictor
 from sam2.sam2_image_predictor import SAM2ImagePredictor
 
 
-_HF_MODEL = "facebook/sam2-hiera-large"
+class Florence2SAM2Config(BaseModel):
+    hf_model: str = Field(
+        default="facebook/sam2-hiera-large",
+        description="Name of the model",
+    )
+    device: Device = Field(
+        default=Device.GPU
+        if torch.cuda.is_available()
+        else Device.MPS
+        if torch.backends.mps.is_available()
+        else Device.CPU,
+        description="Device to run the model on. Options are 'cpu', 'gpu', and 'mps'. Default is the first available GPU.",
+    )
```

Comment:
@hugohonda does this configuration also takes into account the florencev2 config? If not, this config should take into account both model configs (sam2 and florencev2).

Reply:
Hey @CamiloInx ! As florencev2 has no config yet and it is not using the abstraction of SAM2 Class (using it directly) right now it can be independent. But I agree with you